components/utils.py

```diff
@@ -286,9 +287,11 @@ def _generate_ssl_certificate(ips,
     # Remove duplicates from ips
     ips.append('127.0.0.1')
     ips = set(ips)
-    metadata_items = ['IP:{0},DNS:{0}'.format(x) for x in ips]
-    cert_metadata = '{0},DNS:localhost'.format(
-        ','.join(metadata_items))
+    dns_names.append('localhost')
+    dns_names = set(dns_names)
+    metadata_ips = ['IP:{0}'.format(x) for x in ips]
```

Comment:
the reason we used to put the ip addreses as `DNS:` entries is that, some browsers (chrome) basically outright ignore `IP` entries, but work fine when the ip addresses are in `DNS:` entries (surprisingly). (eg. http://www.michaelm.info/blog/?p=1281 - the bug linked there is already closed but AFAIR it did still work like this when I checked this year. I'll check again)

So if this is used to generate a cert that can be seen by browsers (and it is used for the external cert), we should unfortunately still put ips as both `IP:` (for firefox) and `DNS:` entries (for  chrome)

Reply:
@tehasdf to summarize: everything in `ips` should end up in `dns` as well, but not the other way around, correct?

Reply:
yeah, that's right